build fail on musl libc system

Library fail to build on musl libc system.

rust-1.43.1
musl-1.1.24

```
   Compiling devicemapper v0.28.0
     Running `rustc --crate-name devicemapper --edition=2018 /var/tmp/portage/sys-fs/stratisd-2.1.0/work/cargo_home/gentoo/devicemapper-0.28.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C metadata=45f19f9a80ed05ca -C extra-filename=-45f19f9a80ed05ca --out-dir /var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps -L dependency=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps --extern bitflags=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/libbitflags-4ad257329518f4d6.rmeta --extern error_chain=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/liberror_chain-51d2a8a073ff5603.rmeta --extern libc=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/liblibc-793b093d0fe4ab0b.rmeta --extern nix=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/libnix-00d076fb4278826b.rmeta --extern serde=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/libserde-e8f472fd9309df99.rmeta --cap-lints allow -C target-feature=-crt-static -L native=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/build/backtrace-sys-87ba850fde0085c1/out/.libs`
error[E0308]: mismatched types
   --> /var/tmp/portage/sys-fs/stratisd-2.1.0/work/cargo_home/gentoo/devicemapper-0.28.0/src/core/dm.rs:141:78
    |
141 |                 unsafe { convert_ioctl_res!(nix_ioctl(self.file.as_raw_fd(), op, v.as_mut_ptr())) }
    |                                                                              ^^ expected `i32`, found `u64`
    |
help: you can convert an `u64` to `i32` and panic if the converted value wouldn't fit
    |
141 |                 unsafe { convert_ioctl_res!(nix_ioctl(self.file.as_raw_fd(), op.try_into().unwrap(), v.as_mut_ptr())) }
    |                                                                              ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `devicemapper`.

Caused by:
  process didn't exit successfully: `rustc --crate-name devicemapper --edition=2018 /var/tmp/portage/sys-fs/stratisd-2.1.0/work/cargo_home/gentoo/devicemapper-0.28.0/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts --crate-type lib --emit=dep-info,metadata,link -C opt-level=3 -C metadata=45f19f9a80ed05ca -C extra-filename=-45f19f9a80ed05ca --out-dir /var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps -L dependency=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps --extern bitflags=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/libbitflags-4ad257329518f4d6.rmeta --extern error_chain=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/liberror_chain-51d2a8a073ff5603.rmeta --extern libc=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/liblibc-793b093d0fe4ab0b.rmeta --extern nix=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/libnix-00d076fb4278826b.rmeta --extern serde=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/deps/libserde-e8f472fd9309df99.rmeta --cap-lints allow -C target-feature=-crt-static -L native=/var/tmp/portage/sys-fs/stratisd-2.1.0/work/stratisd-2.1.0/target/release/build/backtrace-sys-87ba850fde0085c1/out/.libs` (exit code: 1)
warning: build failed, waiting for other jobs to finish...
```

--- a/src/core/dm.rs
+++ b/src/core/dm.rs
@@ -4,7 +4,7 @@
 
 use std::{cmp, fs::File, mem::size_of, os::unix::io::AsRawFd, slice, u32};
 
-use nix::libc::{c_ulong, ioctl as nix_ioctl};
+use nix::libc::ioctl as nix_ioctl;
 
 use crate::{
     core::{
@@ -136,8 +136,7 @@ impl DM {
                 .as_mut()
                 .expect("pointer to own structure v can not be NULL")
         };
-        let op =
-            request_code_readwrite!(DM_IOCTL, ioctl, size_of::<dmi::Struct_dm_ioctl>()) as c_ulong;
+        let op = request_code_readwrite!(DM_IOCTL, ioctl, size_of::<dmi::Struct_dm_ioctl>());
         loop {
             if let Err(err) =
                 unsafe { convert_ioctl_res!(nix_ioctl(self.file.as_raw_fd(), op, v.as_mut_ptr())) }

